core/src/main/scala/org/apache/spark/deploy/history/HistoryServerDiskManager.scala

```diff
@@ -75,14 +75,32 @@ private class HistoryServerDiskManager(
 
     // Go through the recorded store directories and remove any that may have been removed by
     // external code.
-    val orphans = listing.view(classOf[ApplicationStoreInfo]).asScala.filter { info =>
-      !new File(info.path).exists()
-    }.toSeq
+    val (existing, orphans) = listing
+      .view(classOf[ApplicationStoreInfo])
+      .asScala
+      .toSeq
+      .partition { info =>
+        new File(info.path).exists()
+      }
 
     orphans.foreach { info =>
       listing.delete(info.getClass(), info.path)
     }
 
+    // Reading level db would trigger table file compaction, then it may cause size of level db
+    // directory changed. When service restarts, "currentUsage" is calculated from real directory
+    // size. Update "ApplicationStoreInfo.size" to ensure "currentUsage" equals
+    // sum of "ApplicationStoreInfo.size".
+    val changedStoreInfo = existing.filter { info =>
```

Comment:
Nit: I think we can write this in one loop:
```
existing.foreach { info =>
  val fileSize = sizeOf(new File(info.path)
  if (fileSize != info.size) {
    listing.write(info.copy(size = fileSize))
  }
}
```

Reply:
thanks, updated.